clippy_lints/src/utils/internal_lints/metadata_collector.rs

```diff
@@ -214,6 +258,107 @@ impl Serialize for ApplicabilityInfo {
     }
 }
 
+// ==================================================================
+// Configuration
+// ==================================================================
+#[derive(Debug)]
+pub(crate) struct ClippyConfigurationBasicInfo {
+    pub name: &'static str,
+    pub config_type: &'static str,
+    pub default: &'static str,
+    pub doc_comment: &'static str,
+    pub deprecation_reason: Option<&'static str>,
+}
+
+#[derive(Debug, Clone, Default)]
+struct ClippyConfiguration {
+    name: String,
+    lints: Vec<String>,
+    doc: String,
+    config_type: &'static str,
+    default: String,
+    deprecation_reason: Option<&'static str>,
+}
+
+fn collect_configs() -> Vec<ClippyConfiguration> {
+    let cons = crate::utils::conf::metadata::get_configuration_metadata();
+    cons.iter()
+        .map(move |x| {
+            let (lints, doc) = parse_config_field_doc(x.doc_comment)
+                .unwrap_or_else(|| (vec![], "[ERROR] MALFORMED DOC COMMENT".to_string()));
+
+            ClippyConfiguration {
+                name: to_kebab(x.name),
+                lints,
+                doc,
+                config_type: x.config_type,
+                default: clarify_default(x.default),
+                deprecation_reason: x.deprecation_reason,
+            }
+        })
+        .collect()
+}
+
+fn clarify_default(default: &'static str) -> String {
+    if let Some((_start, init)) = default.split_once('[') {
+        if let Some((init, _end)) = init.split_once(']') {
+            return format!("[{}]", init);
```

Comment:
Needs a comment explaining. This seems risky?

Reply:
This simply strips iterator and struct creation code from the default value. The current python script does the same via a regex. The result can be seen in this example: https://rust-lang.github.io/rust-clippy/master/#blacklisted_name

Documenting that in the code might be better though, fair point^^

What do you exactly mean with "rusky" btw?

Reply:
Risky since there could be multiple [ ] chars, possibly inside a string literal.

Reply:
Could use rsplit_once for the right side.

Reply:
I found that `format("{:?}", default)` will produce a well formatted and readable string for all of our configurations without any adjustments. So, I'll switch over to using that.

Sadly I'm on a different setup for this weekend that for some reason can't compile or even `cargo check` the project. (Some local miss configuration with windows and nightly). I'm currently checking the code in GitHub actions, but it might take some time, or I'll even wait until I'm back at my usual setup.